app/src/contexts/TimeRangeContext.tsx

```diff
@@ -65,52 +67,82 @@ function useTimeRangeMemo(timePreset: TimePreset, timeRangeBounds: TimeRange) {
   const timeRange = useMemo(() => {
     // The timeRangeBounds come from the start / end of the primary dataset
     // Because our time windows are right open (don't include the right time), we need to expand it by a small amount
-    const endTimeBounds = roundToNearestMinutes(
-      endOfHour(addSeconds(timeRangeBounds.end, 1)),
-      { roundingMethod: "floor" }
-    );
-    const startTimeBounds = startOfHour(timeRangeBounds.start);
+    // endOfHourTime is used for hour level time ranges, and endOfDay is used for day level time ranges
+    const paddedEndTimeBounds = addSeconds(timeRangeBounds.end, 1);
```

Comment:
The [backend](https://github.com/Arize-ai/phoenix/blob/a92253fb514b4384e3b709c16612f994e31a3cfe/src/phoenix/core/model_schema.py#L653-L656) is already padded and is on a minute resolution, so we may want to remove the added second here because it may create off-by-one errors with the `endOfX` methods used below, in the rare case where the end time (after being padded by the backend) is at exactly the top of the hour (and adding one second would push it to the next hour or the next day).

Reply:
Oh Interesting, let me remove it and run some tests.